src/errors.rs

```diff
@@ -1,5 +1,21 @@
-//! Error types to use within handlers.
-
+//! # Error Handling
+//!
+//! Restate handles retries for failed invocations.
+//! By default, Restate does infinite retries with an exponential backoff strategy.
+//!
+//! For failures for which you do not want retries, but instead want the invocation to end and the error message
+//! to be propagated back to the caller, you can throw a **terminal error**.
```

Comment:
```suggestion
//! to be propagated back to the caller, you can return a [`TerminalError`].
```

Use links to data types, this is very helpful to navigate throughout the docs.